File: leetcode/easy/94-binaryTreeInorderTraversal.java
class Solution {
    private ArrayList<Integer> values = new ArrayList<>();
    
    public List<Integer> inorderTraversal(TreeNode root) {
        traverse(root);
        return values;
    }
    
    public void traverse(TreeNode node) {
        if (node == null) return;
        traverse(node.left);
        this.values.add(node.val);
        traverse(node.right);
    }
}

File: leetcode/easy/94-binaryTreeInorderTraversal.cpp
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> output;

        traverse(output, root);

        return output;
    }

    void traverse(vector<int> &vect, TreeNode *root) {
        if (root == nullptr) return;

        traverse(vect, root->left);
        vect.push_back(root->val);
        traverse(vect, root->right);
    }
};
